app/main.py

```diff
@@ -1,8 +1,35 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
 
+    def __init__(self, name, age):
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    def set_spouse(self, spouse_name):
+        if spouse_name:
+            spouse = Person.people.get(spouse_name)
+            if spouse:
+                if hasattr(self, "wife") or hasattr(self, "husband"):
```

Comment:
The check `if hasattr(self, "wife") or hasattr(self, "husband"):` is used to prevent setting a spouse if one already exists. However, this logic assumes that a person can only have one spouse attribute, which may not be the case if the logic changes or if there are multiple spouse attributes. Consider using a more robust check or refactoring the logic to handle different scenarios.